File: HackerRank Solutions/Algorithms/Implementation/Find Digits.cpp
#include <bits/stdc++.h>
using namespace std;

// Complete the findDigits function below.
int findDigits(int n) {
    int divisorOfn = 0;
    int t = n;
    while(n > 0){
        int digit = n % 10;
        n = n/10;
        if(digit == 0){
            continue;
        }
        else if(t % digit == 0){
            divisorOfn++;
        }
    }
    return divisorOfn;
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    int t;
    cin >> t;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    for (int t_itr = 0; t_itr < t; t_itr++) {
        int n;
        cin >> n;
        cin.ignore(numeric_limits<streamsize>::max(), '\n');

        int result = findDigits(n);

        fout << result << "\n";
    }

    fout.close();

    return 0;
}

File: HackerRank Solutions/Algorithms/Implementation/Find Digits.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the findDigits function below.
def findDigits(n):
    divisorOfn = 0
    t = n
    while(n > 0):
        digit = n % 10
        n = n//10
        if(digit == 0):
            continue
        if(t % digit == 0):
            divisorOfn += 1
    return divisorOfn
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input())

    for t_itr in range(t):
        n = int(input())

        result = findDigits(n)

        fptr.write(str(result) + '\n')

    fptr.close()
